src/validation/custom.validation.ts

```diff
@@ -78,11 +78,75 @@ export const checkDateValueIsValid = (errMsg: string, dayStr: string = "", month
 
 export const checkOptionalDate = (dayStr: string = "", monthStr: string = "", yearStr: string = "") => {
   if ( dayStr !== "" || monthStr !== "" || yearStr !== "" ) {
-    checkDateValueIsValid(ErrorMessages.INVALID_DATE, dayStr, monthStr, yearStr);
-    checkDateIsInPastOrToday(ErrorMessages.DATE_NOT_IN_PAST_OR_TODAY, dayStr, monthStr, yearStr);
+    const areDateFieldsPresent = checkDateFieldsArePresent(dayStr, monthStr, yearStr);
+    if (areDateFieldsPresent) {
+      const isDateValid = checkDateValueIsValid(ErrorMessages.INVALID_DATE, dayStr, monthStr, yearStr);
+      if (isDateValid) {
+        const isDateInThePast = checkDateIsInPastOrToday(ErrorMessages.DATE_NOT_IN_PAST_OR_TODAY, dayStr, monthStr, yearStr);
+        if (isDateInThePast) {
+          checkDateIsWithinLast3Months(ErrorMessages.IDENTITY_CHECK_DATE_NOT_WITHIN_PAST_3_MONTHS, dayStr, monthStr, yearStr);
+        }
+      }
+    }
+  }
+  return true;
+};
+
+export const checkIdentityDate = (dayStr: string = "", monthStr: string = "", yearStr: string = "") => {
+  const areMandatoryDateFields = checkMandatoryDate(dayStr, monthStr, yearStr);
```

Comment:
Variable name still a bit confusing (to me). How about `isDateValid` as I think that's what the `checkMandatoryDate` function is doing?

`areMandatoryDateFields` makes me think 'are mandatory date fields.... what?

Reply:
Renamed once more.

Reply:
Thanks

Reply:
I was expecting variable name (`areMandatoryDateFields`) and not function name (`checkMandatoryDate`) to change. I think function name is fine - most of the other functions in that file start with 'check...'.

Reply:
Reverted function name back.

Reply:
Thanks.